src/type-exercises.ts

```diff
@@ -0,0 +1,262 @@
+ *
+ * @example
+ * type OmitBoolean = OmitByType<{
+ *   name: string;
+ *   count: number;
+ *   isReadonly: boolean;
+ *   isEnable: boolean;
+ * }, boolean>;
+ *
+ * Resulting type:
+ *
+ * {
+ * name: string;
+ * count: number;
+ * }
+ */
+
+
+// Add here your solution
+const greeting = <T, U>(toFilter: T, toOmmit: U) => {
+    let finalObject: Partial<T> = {};
+    for (const key in toFilter) {
+        if (typeof toFilter[key] !== typeof toOmmit) {
+            finalObject[key] = toFilter[key];
+        }
+    }
+    return finalObject as Partial<T>;
+}
+
+interface customType {
+    name: string,
+    count: number,
+    isReadonly: boolean,
+    isEnable: boolean
+}
+// Add here your example
+console.log('Exercise #1: Filter object properties by type ✨');
+console.log('Ommiting string type from the object');
+console.log(
+    greeting<customType, string>({
+        name: 'test',
+        count: 1,
+        isReadonly: true,
+        isEnable: false
+    }, '')
+);
+
+console.log('Ommiting boolean type from the object');
+console.log(
+    greeting<customType, boolean>({
+        name: 'test',
+        count: 1,
+        isReadonly: true,
+        isEnable: false
+    }, false)
+);
+
+console.log('Ommiting number type from the object');
+console.log(
+    greeting<customType, number>({
+        name: 'test',
+        count: 1,
+        isReadonly: true,
+        isEnable: false
+    }, 0)
+);
+
+/**
+ * Exercise #2: Implement the utility type `If<C, T, F>`, which evaluates a condition `C`
+ * and returns one of two possible types:
+ * - `T` if `C` is `true`
+ * - `F` if `C` is `false`
+ *
+ * @description
+ * - `C` is expected to be either `true` or `false`.
+ * - `T` and `F` can be any type.
+ *
+ * @example
+ * type A = If<true, 'a', 'b'>;  // expected to be 'a'
+ * type B = If<false, 'a', 'b'>; // expected to be 'b'
+ */
+
+// Add here your solution
+type If<C extends boolean, T, F> = C extends true ? T : F;
+
+// Add here your example
+console.log('Exercise #2: Implement the utility type `If<C, T, F>` 👀');
+console.log('C is true');
+type A = If<true, 'a', 1>;
+console.log('Type A:', 'a' as A);
+
+console.log('C is false');
+type B = If<false, 'a', 1>;
+console.log('Type B:', 2 as B);
+
+/**
+ * Exercise #3: Recreate the built-in `Readonly<T>` utility type without using it.
+ *
+ * @description
+ * Constructs a type that makes all properties of `T` readonly.
+ * This means the properties of the resulting type cannot be reassigned.
+ *
+ * @example
+ * interface Todo {
+ *   title: string;
+ *   description: string;
+ * }
+ *
+ * const todo: MyReadonly<Todo> = {
+ *   title: "Hey",
+ *   description: "foobar"
+ * };
+ *
+ * todo.title = "Hello";       // Error: cannot reassign a readonly property
+ * todo.description = "barFoo"; // Error: cannot reassign a readonly property
+ */
+
+
+// Add here your solution
+type MyReadonly<T> = {
+    readonly [K in keyof T]: T[K];
```

Comment:
Read more about the use of `keyof` to better understand what's happening under the hood in here.